deployment/ccip/changeset/deployer_group.go

```diff
@@ -360,25 +481,66 @@ func BuildTimelockPerChain(e deployment.Environment, state CCIPOnChainState) map
 	return timelocksPerChain
 }
 
-func BuildTimelockAddressPerChain(e deployment.Environment, state CCIPOnChainState) map[uint64]string {
+func BuildTimelockAddressPerChain(e deployment.Environment, onchainState CCIPOnChainState) map[uint64]string {
 	addressPerChain := make(map[uint64]string)
 	for _, chain := range e.Chains {
-		addressPerChain[chain.Selector] = state.Chains[chain.Selector].Timelock.Address().Hex()
+		addressPerChain[chain.Selector] = onchainState.Chains[chain.Selector].Timelock.Address().Hex()
 	}
+
+	// TODO: This should come from the Solana chain state which should be enhanced to contain timlock and MCMS address
+	for selector, chain := range e.SolChains {
+		addresses, _ := addressForChain(e, selector)
+		mcmState, _ := state.MaybeLoadMCMSWithTimelockChainStateSolana(chain, addresses)
+		addressPerChain[selector] = mcmsSolana.ContractAddress(mcmState.TimelockProgram, mcmsSolana.PDASeed(mcmState.TimelockSeed))
+	}
+
 	return addressPerChain
 }
 
-func BuildMcmAddressesPerChainByAction(e deployment.Environment, state CCIPOnChainState, mcmCfg *proposalutils.TimelockConfig) (map[uint64]string, error) {
+func BuildMcmAddressesPerChainByAction(e deployment.Environment, onchainState CCIPOnChainState, mcmCfg *proposalutils.TimelockConfig) (map[uint64]string, error) {
 	if mcmCfg == nil {
 		return nil, errors.New("mcm config is nil, cannot get mcms address")
 	}
 	addressPerChain := make(map[uint64]string)
 	for _, chain := range e.Chains {
-		mcmContract, err := mcmCfg.MCMBasedOnAction(state.EVMMCMSStateByChain()[chain.Selector])
+		mcmContract, err := mcmCfg.MCMBasedOnAction(onchainState.EVMMCMSStateByChain()[chain.Selector])
 		if err != nil {
 			return nil, fmt.Errorf("failed to get mcms for action %s: %w", mcmCfg.MCMSAction, err)
 		}
 		addressPerChain[chain.Selector] = mcmContract.Address().Hex()
 	}
+
+	// TODO: This should come from the Solana chain state which should be enhanced to contain timlock and MCMS address
+	for selector, chain := range e.SolChains {
+		addresses, _ := addressForChain(e, selector)
+		mcmState, _ := state.MaybeLoadMCMSWithTimelockChainStateSolana(chain, addresses)
+		var address string
+		switch mcmCfg.MCMSAction {
```

Comment:
can you use this -https://github.com/smartcontractkit/chainlink/blob/develop/deployment/common/proposalutils/propose.go#L34 ?